File: Backend/API/Models/Reservations.Models/Dto/Filters.cs
using System;

namespace Models.Reservations.Models.Dto
{
    public class Filters
    {
        public string Name { get; set; }
        public DateTime? Day { get; set; }
        public DateTime? StartAfter { get; set; }
        public DateTime? StartUntil { get; set; }
        public DateTime? EndAfter { get; set; }
        public DateTime? EndUntil { get; set; }
        public int? TableNumber { get; set; }
    }
}

File: reservation-web/src/app/reservations/models/filters.ts
export class Filters {
    public name: string;
    public day: Date;
    public startAfter: Date;
    public startUntil: Date;
    public endAfter: Date;
    public endUntil: Date;
    public realStartAfter: Date;
    public realStartUntil: Date;
    public realEndAfter: Date;
    public realEndUntil: Date;
    public tableNumber: number;

    constructor(data?: Partial<Filters>) {
        Object.assign(this, data);
    }
}
